### ISSUE TITLE
minimum required pytest version

### ISSUE THREAD

**Issue by alfredodeza** [jaraco/jaraco.functools#13]
I'm trying to package cherrypy for EPEL8 and one of the many dependencies is this tool which requires `pytest>=3.5` (as shown in https://github.com/jaraco/jaraco.functools/commit/021b18b89cf83977397350ebe54603032086baf6) 

The commit doesn't explain why, I believe this might be because `pytest-flake8` requires it? Since `pytest` doesn't have a `3.5` packaged, it would mean that we need to get that working in order to package `jaraco.functools`.

Could you help clarify why `3.5` is required or if it is at all possible to use any older version?

**Comment by ktdreyer** [comment]
We're able to modify this back down to `pytest>=3.4` without issue for `jaraco.classes`, so we're wondering if we can do that for each of the CherryPy dependencies.

**Comment by ktdreyer** [comment]
@alfredodeza pointed out to me that `jaraco.classes` is only running flake8 during pytest, whoops.

**Comment by jaraco** [comment]
Hey Alfredo. Sorry I let this one linger so long. Nothing personal, just lost in the noise.

That commit exists in jaraco/skeleton@021b18b, shared by most of the [projects I maintain](https://pypi.org/user/jaraco). Based on the commit message, I suspect I bumped the version number just to set some reasonable default. Although, looking at the [changelog for that release](http://doc.pytest.org/en/latest/changelog.html#pytest-3-5-0-2018-03-21), I see features I've definitely used in some projects or executions:

- New `--show-capture command-line` option that allows to specify how to display captured output when tests fail: no, stdout, stderr, log or all (the default). (`#1478`)
- Fixtures are now instantiated based on their scopes, with higher-scoped fixtures (such as session) being instantiated first than lower-scoped fixtures (such as function). The relative order of fixtures of the same scope is kept unchanged, based in their declaration order and their dependencies. (`#2405`)
- Add command line option --deselect to allow deselection of individual tests at collection time. (`#3198`)
- Captured logs are printed before entering pdb. (`#3204`)
- The builtin module platform is now available for use in expressions in pytest.mark. (`#3236`)

Probably there are other features and behaviors that the projects rely on that were released after 3.5, but 3.5 seems like a reasonably-conservative lower bound, especially considering that it was released over 2 years ago.

So it would be difficult to change this as the default or even just for this project.
